### ISSUE TITLE
functools.partial allows overwriting an already provided keyword at call time

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#2777]
**Describe the bug**
Might be a not well documented "feature" of `functools.partial`: it is possible to overwrite an already provided keyword of a partial function by providing the keyword argument at call time `partial(test, x="a")(x="b")`. This does not work with positional arguments.

Pyright doesn't seem to be aware that keyword arguments can be provided a second time at call time.

**To Reproduce**
```py
from functools import partial
from typing import Protocol

class Test(Protocol):
    def __call__(self, x: str = ...) -> str:
        ...

def test(x: str = "") -> str:
    return x

fun: Test = partial(test, x="a")
print(fun(x="b"))  # prints "b"
```
```
error: Expression of type "partial[str]" cannot be assigned to declared type "Test"
    "partial[str]" is incompatible with protocol "Test"
      "__call__" is an incompatible type
        Type "() -> str" cannot be assigned to type "(x: str = ...) -> str"
          Function accepts too many positional parameters; expected 0 but received 1 (reportGeneralTypeIssues)
```

**Comment by erictraut** [comment]
Thanks for the bug report. Yeah, the documentation no only doesn't mention this behavior, but it seems to indicate that the keyword parameters are "frozen" by the initial call to `partial`.

I've updated the logic in pyright to reflect this undocumented behavior.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.203, which I just published. It will also be included in the next release of pylance.
